RiotSwiftUI/Modules/Room/LocationSharing/LocationAnnotation.swift

```diff
@@ -17,26 +17,33 @@
 import Foundation
 import Mapbox
 
-class UserLocationAnnotation: NSObject, MGLAnnotation {
+class LocationAnnotation: NSObject, MGLAnnotation {
     
     // MARK: - Properties
     
-    let avatarData: AvatarInputProtocol
-    
     let coordinate: CLLocationCoordinate2D
     
-    let coordinateType: LocationSharingCoordinateType
-    
     // MARK: - Setup
     
-    init(avatarData: AvatarInputProtocol,
-         coordinate: CLLocationCoordinate2D,
-         coordinateType: LocationSharingCoordinateType) {
+    init(coordinate: CLLocationCoordinate2D) {
         
         self.coordinate = coordinate
+    }
+}
+
```

Comment:
You have to create the intermediate `class PinLocationAnnotation: LocationAnnotation`. `UserLocationAnnotation` can be casted as `LocationAnnotation` and if we add a new kind of annotation like `POIAnnotation` we will not be able to distinct them.